### PR TITLE
PMM-7485 PMM-7359 Dependency Graph and default version

### PR DESCRIPTION
**Pull request by BupycHuk**
[PMM-7485](https://jira.percona.com/browse/PMM-7485)
[PMM-7359](https://jira.percona.com/browse/PMM-7359)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1625

- [ ] Links to other linked pull requests (optional).
- https://github.com/percona/pmm/pull/712
- https://github.com/percona-platform/dbaas-controller/pull/163
- https://github.com/percona-platform/dbaas-api/pull/112
- https://github.com/percona/pmm-managed/pull/706
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: services/management/dbaas/version_service_client.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		if params.dbVersion != "" {
+			paths = append(paths, params.dbVersion)
+		}
+	}
+	url := strings.Join(paths, "/")
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := c.http.Do(req)
```
[the remark this answers is not in the document]
Comment (jprukner) [reply services/management/dbaas/version_service_client.go:126]: @BupycHuk please add `defer resp.Body.Close()`

[the remark this answers is not in the document]
Comment (BupycHuk) [reply services/management/dbaas/version_service_client.go:126]: wow, how could I do this I had in mind that I should close the body.

**Review by jprukner — APPROVED** [review]
Looks good to me.

**Review by askomorokhov — APPROVED** [review]
Looks good to me.
Missed a few docstrings for public methods.

File: services/management/dbaas/components_service.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+func NewComponentsService(db *reform.DB, dbaasClient dbaasClient, versionServiceClient versionService) dbaasv1beta1.ComponentsServer {
+	l := logrus.WithField("component", "components_service")
+	return &componentsService{
+		l:                    l,
+		db:                   db,
+		dbaasClient:          dbaasClient,
+		versionServiceClient: versionServiceClient,
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply services/management/dbaas/components_service.go:47]: good question

**Pull request merged**